sdks/python/apache_beam/transforms/display_test.py

```diff
@@ -22,12 +22,76 @@
 from datetime import datetime
 import unittest
 
+import hamcrest as hc
+from hamcrest.core.base_matcher import BaseMatcher
+
 import apache_beam as beam
 from apache_beam.transforms.display import HasDisplayData
 from apache_beam.transforms.display import DisplayData
 from apache_beam.transforms.display import DisplayDataItem
 
 
+def make_nspace_display_data(component):
+  dd = DisplayData.create_from(component)
+  nspace = '{}.{}'.format(component.__module__, component.__class__.__name__)
+  return nspace, dd
+
+
+class ItemMatcher(BaseMatcher):
+  def __init__(self, key, value, namespace, match_props=None):
+    self.key = key
+    self.value = value
+    self.namespace = namespace
+
+    if not match_props:
+      raise ValueError('Must match at least one attribute.')
+    self._match_props = match_props
+
+  def _matches(self, item):
+    if item.key != self.key and 'key' in self._match_props:
+      return False
+    if item.namespace != self.namespace and 'namespace' in self._match_props:
+      return False
+    if item.value != self.value and 'value' in self._match_props:
+      return False
+    return True
+
+  def describe_to(self, description):
+    if 'key' in self._match_props:
+      description.append('key is {} '.format(self.key))
+    if 'value' in self._match_props:
+      description.append('value is {} '.format(self.value))
+    if 'namespace' in self._match_props:
+      description.append('namespace is {} '.format(self.namespace))
+
+  @classmethod
+  def matches_key(cls, key):
```

Comment:
So far I only see usage of `matches_kvn`. I recommend removing the additional APIs you aren't using-- it's easier to start with a small API and grow it later than to start with a wide API and narrow it.